api/src/test/java/org/openmrs/api/UserServiceTest.java

```diff
@@ -353,7 +350,7 @@ public void createUser_shouldNotAllowAssigningSuperUserRoleIfCurrentUserDoesnotH
 			u.addRole(role);
 			// here we expect the exception to be thrown
 			userService.createUser(u, "Openmr5xy");
-		});
```

Comment:
This commit does way too much, which is why the test fails.

* The point of wrapping things in the `withCurrentUserAs()` function is that this enables us to set the current executing user to a particular user. By default, the executing user for these tests is "admin", so this test won't fail as expected.
* The call to `role.hasPrivilege(PrivilegeConstants.ASSIGN_SYSTEM_DEVELOPER_ROLE);` doesn't do anything at all. Or rather, it checks to see if the role has that privilege but by itself, that won't throw the error we're expecting, which happens in the `checkSuperUserPrivilege()` code.

My advice would be to undo this last commit and make a much simpler commit that simply changes this line:

```java
expectedException.expectMessage("User you must have the following privileges in order to assign them: {0}"); 
```

To this:

```java
expectedException.expectMessage("You must have the role {0} in order to assign it"); 
```

You can undo the last commit by running: `git reset --hard HEAD~1`.

Reply:
ok then let me do that